pkg/utils/gateway_wrapper.go

```diff
@@ -0,0 +1,80 @@
+package utils
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	MultiClusterIPAddressType       gatewayv1beta1.AddressType = "kuadrant.io/MultiClusterIPAddress"
+	MultiClusterHostnameAddressType gatewayv1beta1.AddressType = "kuadrant.io/MultiClusterHostnameAddress"
+)
+
+type GatewayWrapper struct {
+	*gatewayv1beta1.Gateway
+	isMultiCluster bool
+}
+
+func NewGatewayWrapper(g *gatewayv1beta1.Gateway) (*GatewayWrapper, error) {
+	gw := &GatewayWrapper{Gateway: g, isMultiCluster: false}
+
+	for i, address := range gw.Status.Addresses {
+		if i == 0 {
+			gw.isMultiCluster = isMultiClusterAddressType(*address.Type)
+			continue
+		}
+		if gw.isMultiCluster == isMultiClusterAddressType(*address.Type) {
+			continue
+		}
+		return nil, fmt.Errorf("gateway is invalid: inconsistent status addresses")
+
+	}
+	return gw, nil
+}
+
+func isMultiClusterAddressType(addressType gatewayv1beta1.AddressType) bool {
+	return addressType == MultiClusterIPAddressType || addressType == MultiClusterHostnameAddressType
+}
+
+func (a *GatewayWrapper) GetKind() string {
+	return "GatewayWrapper"
+}
+
+func (a *GatewayWrapper) IsMultiCluster() bool {
+	return a.isMultiCluster
+}
+
+func (a *GatewayWrapper) GetNamespaceName() types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: a.Namespace,
+		Name:      a.Name,
+	}
+}
+
+func (a *GatewayWrapper) String() string {
+	return fmt.Sprintf("kind: %v, namespace/name: %v", a.GetKind(), a.GetNamespaceName())
+}
```

Comment:
I think we could remove `GetKind()`, `GetNamespaceName()`, and  `String()`, not sure we need any of that.

Reply:
got rid of them